x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanStreamOutput.java

```diff
@@ -226,6 +226,17 @@ public void writeCachedString(String string) throws IOException {
         writeString(string);
     }
 
+    @Override
+    public void writeOptionalCachedString(String str) throws IOException {
+        if (str == null) {
+            writeBoolean(false);
+            return;
+        }
+
+        writeBoolean(true);
+        writeCachedString(str);
```

Comment:
@luigidellaquila , this internally performs a transport version check

```
        if (getTransportVersion().before(TransportVersions.ESQL_CACHED_STRING_SERIALIZATION)) {
            writeString(string);
...
```
But that's not ideal since `writeOptionalCachedString` itself will be used only if the TV is greater than the new TV that this PR introduces.

I think we should move the TV check that's currently inside `writeCachedString` _out_ into the consumers of that method.

If you agree, I'd perform that little refactor as part of this PR.

Reply:
> that little refactor

2c: there are more than a dozen `writeCachedString()` consumers and only this call here would perform a double check.

Reply:
Fair enough, let's not have so much code duplication by moving the transport version check into every consumer.

But! We can have our cake and eat it, too. I think I just wanna refactor `writeOptionalCachedString` so it doesn't perform a redundant second transport version check, by moving the logic from `writeCachedString` that's for newer nodes, only, into a separate method.

Otherwise, the serialization changes to FieldAttribute may become confusing overall.